sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -155,16 +155,35 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
 
         public static AmqpMessage SBMessageToAmqpMessage(ServiceBusMessage sbMessage)
         {
+            // body
             var amqpMessage = sbMessage.ToAmqpMessage();
+
+            // properties
             amqpMessage.Properties.MessageId = sbMessage.MessageId;
             amqpMessage.Properties.CorrelationId = sbMessage.CorrelationId;
             amqpMessage.Properties.ContentType = sbMessage.ContentType;
+            amqpMessage.Properties.ContentEncoding = sbMessage.AmqpMessage.Properties.ContentEncoding;
             amqpMessage.Properties.Subject = sbMessage.Subject;
             amqpMessage.Properties.To = sbMessage.To;
             amqpMessage.Properties.ReplyTo = sbMessage.ReplyTo;
             amqpMessage.Properties.GroupId = sbMessage.SessionId;
             amqpMessage.Properties.ReplyToGroupId = sbMessage.ReplyToSessionId;
+            amqpMessage.Properties.GroupSequence = sbMessage.AmqpMessage.Properties.GroupSequence;
 
+            if (sbMessage.AmqpMessage.Properties.UserId.HasValue)
+            {
+                ReadOnlyMemory<byte> userId = sbMessage.AmqpMessage.Properties.UserId.Value;
+                if (MemoryMarshal.TryGetArray(userId, out ArraySegment<byte> segment))
+                {
+                    amqpMessage.Properties.UserId = segment;
+                }
+                else
+                {
+                    amqpMessage.Properties.UserId = new ArraySegment<byte>(userId.ToArray());
+                }
+            }
+
+            // If TTL is set, it is used to calculate AbsoluteExpiryTime and CreationTime
```

Comment:
This is the existing logic that we copied forward from T1. It is debatable whether this is the desired behavior now that users can set these properties, but changing it would be a breaking change.

Reply:
Actually, I take this back. Digging through old emails it looks like we intentionally left this behavior because the service will still populate the AbsoluteExpiryTime even if the user doesn't set it.